### PR TITLE
[Batch] March API changes

### PR DESCRIPTION
**Pull request by bgklein**

### CODE REVIEW

**Comment by g2vinay** [comment]
@bgklein 
Is the Batch SDK moving towards our Track 2 Guidelines?
are you expecting the azure-sdk-for-java team to review this ?
Is this PR blocking any release of Batch SDK?
Plus, the CI is failing below too.

cc: @JonathanGiles , @joshfree

**Comment by bgklein** [comment]
We aren't waiting on you guys :) Just @xingwu1 for review. Ci was failing due to missing pom.xml updates

**Comment by bgklein** [comment]
Actually @g2vinay, you guys may want to check the testing changes. A while back you guys asked us to create the pools in a VNet since you were live testing our CI instead of using recordings (and it was a security requirement for long lived pools to be in a VNet), so you turned off our testing.

Additionally due to this we now have a test dependency on azure-mgmt-network. Do you know where we have to declare this as an external script to pass the CI script as that appears to be what is failing it.

**Comment by g2vinay** [comment]
@bgklein 
I believe you need to add the dependency to eng/versioning/external_dependencies.txt
@JimSuplizio , please  correct me I am wrong

File: sdk/batch/microsoft-azure-batch/pom.xml [hunk 1]
```diff
@@ -67,6 +67,19 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>com.microsoft.azure</groupId>
```
Comment (JimSuplizio) [inline sdk/batch/microsoft-azure-batch/pom.xml:71]: @g2vinay, thank you for the @ mention. 

@bgklein you've added the dependencies and the correct tags but neither com.microsoft.azure:azure nor com.microsoft.azure:azure-mgmt-network have the appropriate entries in [eng/versioning/external_dependencies.txt](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/versioning/external_dependencies.txt) which is the reason that the Analyze step is failing. You're going to want to add the following entries to the file. Note that com.microsoft.azure:azure-mgmt-network is a test only dependency and should be added under the test dependency versions. In both cases please add them alphabetically. 
com.microsoft.azure:azure:1.24.1
com.microsoft.azure:azure-mgmt-network:1.33.0

**Comment by JimSuplizio** [comment]
I did not mean to approve, I meant to request changes which are detailed in the review.

File: eng/versioning/external_dependencies.txt [hunk 2]
```diff
@@ -54,9 +54,12 @@ org.slf4j:slf4j-simple;1.7.25
 cglib:cglib-nodep;3.2.7
 com.github.tomakehurst:wiremock-standalone;2.24.1
 com.microsoft.azure:adal4j;1.6.5
+com.microsoft.azure:azure;1.24.1
 com.microsoft.azure:azure-mgmt-graph-rbac;1.3.0
 com.microsoft.azure:azure-mgmt-keyvault;1.11.1
+com.microsoft.azure:azure-mgmt-network:1.33.0
```
Comment (JimSuplizio) [inline eng/versioning/external_dependencies.txt:60]: @bgklein blarg, I'm sorry, there was a typo in the feedback, the last colon needs to be a semicolon. com.microsoft.azure:azure-mgmt-network:1.33.0 should be
com.microsoft.azure:azure-mgmt-network;1.33.0

**Comment by bgklein** [comment]
I'll merge and release this Monday. On-call this weekend/today so likely won't have time today

**Pull request merged**